rollup/rcfg/config.go

```diff
@@ -39,5 +39,13 @@ var (
+	// CurieL1GasPriceOracleBytecode is the new (blob-enabled) gas price oracle after
+	// the Curie hard fork. Run these commands in the monorepo to verify this bytecode:
+	//
+	// git checkout 7059ad0ed465201287e06625dd4058c8ae9682b8
+	// cd contracts
+	// yarn
+	// FOUNDRY_EVM_VERSION="cancun" forge build
```

Comment:
if we use cancun, would the compiler generate some ops incompatible with l2geth?

Just a double-check, I think not because the contract is very unlikely to generate new opcodes introduced in cancun, furthermore, l2geth has supported other opcodes except blob-related ones.

Reply:
> l2geth has supported other opcodes except blob-related ones

It does not support mcopy and tstore/tload (yet).

> Would the compiler generate some ops incompatible with l2geth?

That's a valid concern. The answer for solc 0.8.24 is no, see here: "[EIP-5656](https://eips.ethereum.org/EIPS/eip-5656) finally introduces the MCOPY opcode, available in Yul as the mcopy() builtin. Note that Solidity 0.8.24 only makes it available for explicit use in assembly".

cc @zimpha to confirm.

Reply:
yeah, `mcopy` won't be used unless it is explicitly defined in contracts.

Reply:
got it. btw, I think SetCode does not do the op validity check like create or create2 ops here:
https://github.com/scroll-tech/go-ethereum/blob/scroll-v5.3.27/core/vm/interpreter.go#L200-L206

my original concern is to make sure not to include any invalid ops even after curie, and the answer is no.

Reply:
Right. `SetCode` does not check any opcode anyway, i.e. you can deploy any contract, even one with invalid opcodes.

We considered to check it when we disabled `SELFDESTRUCT` two years ago. But it's tricky, since you don't know which part of the bytecode is code and which is data.

Reply:
makes sense. not known how hard the problem is, to implement such a bytecode parser.